packages/salesforcedx-sobjects-faux-generator/src/describe/sObjectDescribe.ts

```diff
@@ -289,12 +288,28 @@ export class SObjectDescribe {
     nextToProcess: number
   ): Promise<SObject[]> {
     try {
+      const org = await Org.create({
+        aliasOrUsername: await ConfigUtil.getUsername(projectPath)
+      });
+
       if (!this.accessToken || !this.instanceUrl) {
-        await this.getConnectionData(projectPath);
+        await this.getConnectionData(org);
+      }
+
+      let response: XHRResponse;
```

Comment:
With this code there's still the potential of ending up in the same invalid state if refreshing the connection doesn't work. We can change this to be in a "retry function" that will run the call outs two times, at most, and handles errors.

Reply:
If the refresh fails then something is likely wrong with the connection, or the refresh token is bad. We could handle the latter case to clearly ask to reauthorize the org in the error that gets thrown. This was meant to only handle the unauthorized status of the first call and not necessarily be a retry x times kind of thing.

Reply:
I see, let's make sure we handle any errors after doing the refresh then.